runtime/lua/vim/filetype.lua

```diff
@@ -2213,6 +2186,19 @@ local pattern = {
   end,
   -- END PATTERN
 }
+
+-- BEGIN ENV_PATTERN
+local env_pattern = {
+  ['/cabal%.config'] = { 'cabalconfig', 'HOME' },
+  ['/git/attributes'] = { 'gitattributes', 'XDG_CONFIG_HOME' },
+  ['/git/config'] = { 'gitconfig', 'XDG_CONFIG_HOME' },
+  ['/git/ignore'] = { 'gitignore', 'XDG_CONFIG_HOME' },
+  ['/options'] = { 'gpg', 'GNUPGHOME' },
+  ['/gpg%.conf'] = { 'gpg', 'GNUPGHOME' },
+  ['/doc/.*%.txt'] = { 'help', 'VIMRUNTIME' },
+}
```

Comment:
Hmm, can we work with placeholders here in the key instead a second entry? This might conflict with the `string, function` assumption further down the line...